Source/Eto.Gtk/Forms/GtkWindow.cs

```diff
@@ -249,7 +249,10 @@ public override void AttachEvent(string id)
 					Control.DeleteEvent += Connector.HandleDeleteEvent;
 					break;
 				case Eto.Forms.Control.ShownEvent:
-					Control.Shown += Connector.HandleShown;
+					Control.Shown += Connector.HandleShownEvent;
+					break;
+				case Window.OwnerChangedEvent:
```

Comment:
This shouldn't be necessary, this doesn't need to be a HandlerEvent..  I believe this will cause the event to be called more than once, since you are already calling it when setting the Window.Owner property..